src/Promitor.Integrations.AzureMonitor/HttpPipelinePolicies/RecordArmRateLimitMetricsPolicy.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using GuardNet;
+using Promitor.Core;
+using Promitor.Core.Metrics.Interfaces;
+using Promitor.Core.Metrics.Sinks;
+using Promitor.Integrations.Azure.Authentication;
+
+namespace Promitor.Integrations.AzureMonitor.HttpPipelinePolicies{
```

Comment:
These HttpPipelinePolicies are to replicate the legacy SDK behavior, where User-Agent header was sent before sending a request(presumably to track usage?) and ARM rate limiting metrics are recorded after response is received